android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/DateTimeExtension.kt

```diff
@@ -63,6 +63,13 @@ fun Date?.makeItReadable(): String {
 fun Date?.prettifyDate(): String =
   if (this == null) "" else PrettyTime(Locale.getDefault()).formatDuration(this)
 
+fun isSameMonthYear(monthYear: String, yearMonth: String): Boolean {
+  val firstDate = monthYear.parseDate(SDF_MMM_YYYY)?.formatDate(SDF_MMM_YYYY)
+  val secondDate = yearMonth.parseDate(SDF_YYYY_MMM)?.formatDate(SDF_MMM_YYYY)
+  if (firstDate.isNullOrEmpty() || secondDate.isNullOrEmpty()) return false
+  return firstDate == secondDate
```

Comment:
This implementation is generally comparing two strings, I do not think the date parsing matters. You can use the compare function from `StringExtensions.kt `file

Reply:
@ellykits  this method already available in StringExtensions.kt. and i am not using this one though I agree dates should not be compared like this. I am refactoring this function as this is not being used anywhere in the code.